### PR TITLE
FIM System Tests: Improve verify scripts logging and generate global result files

### LINKED ISSUE #586: FIM System Tests: Improve verify scripts logging and generate global result files
Hi all,

The scripts that verify the alerts of the System tests logging can be improved in order to give more details of the process as well as some metrics.

#### verify_alerts_json.py

- [x] Review logging and ensure every log contains proper format: timestamp, logging level, etc...

- [x] Check logged information and add any other information that could be helpful

- [x] Make missing_alerts unique for every host.

- [x] Make verification log unique for each tuple of (scenario, action, host)

#### verify_alerts_elasticsearch.py

- [x] Review logging and ensure every log contains proper format: timestamp, logging level, etc...

- [x] Check logged information and add any other information that could be helpful

- [x] Make missing_alerts unique for every host

- [x] Make verification log unique for each tuple of (scenario, action, host)

#### Results files generation

- [x] Add the corresponding function to generate a JSON:
    - The generated JSON file should include the results of the test for each agent host for both JSON and ES testing.

- [x] Adapt the verification scripts to call the function created in the previous step. The scripts are:
   - [x] `verify_alerts_elasticsearch.py`
      - [x] Add the corresponding arguments.
      - [x] Adapt the code globally.
   - [x] `verify_alerts_json.py`
      - [x] Add the corresponding arguments.
      - [x] Adapt the code globally.
   - [x] Adapt the Ansible tasks file `verify_alerts.yml`

- [x] Add the required variables. 

Best regards,

Jose

### PR DESCRIPTION
**Pull request by rshad**
Hi all!

This PR resolves #586.

Kr,

Rshad

### CODE REVIEW

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py [hunk 1]
```diff
@@ -438,27 +480,43 @@ def verify_es_alerts(files_list, max_retry, query, no_alert_style, es, index_nam
...
-    )
\ No newline at end of file
+    )
+
+    try:
+        assert failure == 0
+    except:
+        logging.error("Verification result is FAILED. Number of failed paths: {}/{}"\
+            .format(failure, expected_alerts_num))
+    finally:
```
Comment (xr09) [inline tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: Regular program logic should not be inside the `finally` clause. Use this only for closing/releasing resources.

File: tests/system/fim/common_tasks/verify_alerts_json.py [hunk 2]
```diff
@@ -183,41 +224,73 @@ def main():
...
+        
+
+        expected_alerts_num = len(paths_list_set)
+        received_alerts_num = expected_alerts_num - len(sub_paths)
 
+        return returb_code
     except Exception:
         logging.critical("An error has ocurred. Exiting")
         raise Exception
+    finally:
```
Comment (xr09) [inline tests/system/fim/common_tasks/verify_alerts_json.py]: Same issue about finally clause.

File: tests/system/fim/common_tasks/verify_alerts_json.py [hunk 3]
```diff
@@ -183,41 +224,73 @@ def main():
...
-            elapsed = time.time() - start
-            logging.info("Elapsed time: ~ %s seconds \n" % int(elapsed))
+            
+            logging.info("Elapsed time: %s" % (elapsed))
+        
+
+        expected_alerts_num = len(paths_list_set)
+        received_alerts_num = expected_alerts_num - len(sub_paths)
 
+        return returb_code
```
Comment (xr09) [inline tests/system/fim/common_tasks/verify_alerts_json.py]: Typo for return_code?

File: tests/system/fim/common_tasks/verify_alerts_elasticsearch.py (hunk 1 above)
Comment (rshad) [reply tests/system/fim/common_tasks/verify_alerts_elasticsearch.py]: Hi @xr09 !

I think including the results files generation at `finally` is a secure way to ensure that the result file will always be generated however the script fails at some point.

Thank you,

Rshad

File: tests/system/fim/common_tasks/verify_alerts_json.py (hunk 2 above)
Comment (rshad) [reply tests/system/fim/common_tasks/verify_alerts_json.py]: Same argument as the mentioned before.

**Pull request merged**